clients/js/test/collect.test.ts

```diff
@@ -110,6 +113,222 @@ test.serial('it can collect', async (t) => {
   t.deepEqual(subtractAmounts(balEnd2, balStart2), sol(0.0015 / 2));
 });
 
+test.serial('it can collect from an asset with plugins', async (t) => {
+  const umi = await createUmi();
+  const asset = await createAsset(umi, {
+    plugins: [
+      {
+        type: 'FreezeDelegate',
+        frozen: false,
+      },
+      {
+        type: 'Attributes',
+        attributeList: [
+          {
+            key: 'Test',
+            value: 'Test',
+          },
+        ],
+      },
+    ],
+  });
+  const balStart1 = await umi.rpc.getBalance(recipient1);
+  const balStart2 = await umi.rpc.getBalance(recipient2);
+  await collect(umi, {})
+    .addRemainingAccounts({
+      isSigner: false,
+      isWritable: true,
+      pubkey: asset.publicKey,
+    })
+    .sendAndConfirm(umi);
+  const balEnd1 = await umi.rpc.getBalance(recipient1);
+  const balEnd2 = await umi.rpc.getBalance(recipient2);
+  t.is(await hasCollectAmount(umi, asset.publicKey), false);
+  t.deepEqual(subtractAmounts(balEnd1, balStart1), sol(0.0015 / 2));
+  t.deepEqual(subtractAmounts(balEnd2, balStart2), sol(0.0015 / 2));
```

Comment:
Nit: these tests are good but I think the check for `hasCollectAmount(false)` is a bit narrow.  Because really what should be checked imo is that it only has the rent amount after the `collect`.  For example, with a slight math error and there was like 0.0005 SOL extra on the account, this test could erroneously pass.

wdyt about calling `hasCollectAmount` right after the asset is created, and having a different helper called hasOnlyRentAmount` called after the collect?

Reply:
Yup, that makes sense.